src/lib/build_state_map.ts

```diff
@@ -67,11 +67,14 @@ export const build_state_map = (
     }
     if (value.json.Type === "Parallel") {
       // update the child states
-      const branches: Array<Record<string, unknown>> = JSONPath({
+      const result:
+        | Array<Record<string, unknown>>
+        | Array<Record<string, unknown>>[] = JSONPath({
         json: definition,
         path: `$..['States']['${key}'].Branches`,
         wrap: false,
       });
+      const branches = Array.isArray(result[0]) ? result[0] : result;
```

Comment:
in the existing test case for parallel, the JSONPath expression returns an array of branches. In the test case submitted with this PR that was causing it to generate incorrectly, we got back an extra Array wrapper. 

I updated the return type definition to indicate the possible presence of the extra wrapper and conditionally unwrap it.